tests/test_template_types.py

```diff
@@ -366,3 +421,478 @@ def test_templates_handle_html(
+    },
+    {
+        'Field number': '34',
+        'Field name': 'NOTIFICATION-BODY',
+        'Mandatory': '',
+        'Data type': 'A',
+        'Comment': """
+            Difficult to define a maximum length as dependent on
+            other formatting factors absolute maximum for OSG is 6
+            pages but ideally no more than 4 pages total. OSG to
+            confirm approach to mark up and line breaks...
+        """,
+        'Example': (
+            'Dear Henry Hadlow,<cr><cr>'
+            'Thank you for applying to register a lasting power of '
+            'attorney (LPA) for property and financial affairs. We '
+            'have checked your application and...<cr><cr>'
+        ),
+    }
+]
+
+
+@freeze_time("2016-04-29 12:00:00.000000")
+@pytest.mark.parametrize('field', dvla_file_spec)
+@pytest.mark.parametrize('template', [
+    LetterDVLATemplate(
+        {
+            "content": (
+                'Dear ((name)),\n\n'
+                'Thank you for applying to register a lasting power of '
+                'attorney (LPA) for property and financial affairs. We '
+                'have checked your application and...'
+            ),
+            'subject': 'Your ((thing)) is due soon',
+        },
+        {
+            'thing': 'application',
+            'name': 'Henry Hadlow',
+            'address line 1': 'Mr Henry Hadlow',
+            'address line 2': 'Managing Director',
+            'address line 3': '123 Electric Avenue',
+            'address line 4': 'Great Yarmouth',
+            'address line 5': 'Norfolk',
+            'address line 6': '',
+            'postcode': 'NR1 5PQ',
+        },
+        numeric_id=1,
+    )
+])
+def test_letter_output_template(field, template):
+    # To debug this test it’s helpful uncomment the following line:
+    # print(field)
+    assert str(template).split('|')[int(field['Field number']) - 1] == field['Example']
+
+
+@pytest.mark.parametrize('template', [
+    LetterDVLATemplate(
+        {
+            "content": 'Pipes | pipes | everywhere',
+            'subject': 'Your | is due soon',
+        },
+        {
+            'thing': '|',
+            'name': '|',
+            'address line 1': '|',
+            'address line 2': 'Managing Director',
+            'address line 3': '123 Electric Avenue',
+            'address line 4': 'Great Yarmouth',
+            'address line 5': 'Norfolk',
+            'address line 6': '',
+            'postcode': 'NR1 5PQ',
+        },
+        numeric_id=1,
+    )
+])
+def test_letter_output_pipe_delimiting(template):
+
+    assert len(str(template).split('|')) == len(dvla_file_spec)
+    assert 'Pipes  pipes  everywhere' in str(template)
+
+
+@pytest.mark.parametrize('id, expected_exception', [
+    (None, 'numeric_id is required'),
+    ('abc123', 'numeric_id must be a number'),
+    (12345678, 'numeric_id cannot be longer than 7 digits'),
+    pytest.mark.xfail((1234567, 'numeric_id cannot be longer than 7 digits')),
```

Comment:
i'm not a huge fan of this test being in the same parametrize stuff, because it's not apparent whether this'll xfail because it doesn't raise a TypeError, or because it does raise a TypeError but the error string is different.